packages/language-server/src/manifest-handling.ts

```diff
@@ -0,0 +1,100 @@
+import { maxBy, map, filter } from "lodash";
+import { readFile } from "fs-extra";
+import { URI } from "vscode-uri";
+import { dirname } from "path";
+import globby from "globby";
+import { FileChangeType } from "vscode-languageserver";
+
+type AbsolutePath = string;
+type ManifestData = Record<AbsolutePath, { isFlexEnabled: boolean }>;
+const manifestData: ManifestData = Object.create(null);
+let manifestDocuments: string[];
+
+export function isManifestDoc(uri: string): boolean {
+  return uri.endsWith("manifest.json");
+}
+
+export async function initializeManifestDocuments(
+  workspaceFolderPath: string
+): Promise<void[]> {
+  manifestDocuments = await findAllManifestDocumentsInWorkspace(
+    workspaceFolderPath
+  );
+  const readManifestPromises = map(manifestDocuments, async (manifestDoc) => {
+    const isFlexEnabled = await readFlexEnabledFlagFromManifestFile(
+      manifestDoc
+    );
+    // Parsing of manifest.json failed because the file is invalid
+    if (isFlexEnabled !== "INVALID") {
+      manifestData[manifestDoc] = { isFlexEnabled };
+    }
+  });
+
+  return Promise.all(readManifestPromises);
+}
+
+export function getFlexEnabledFlagForXMLFile(xmlPath: string): boolean {
+  const manifestFilesForCurrentFolder = filter(
+    Object.keys(manifestData),
+    (manifestPath) => xmlPath.includes(dirname(manifestPath))
+  );
+  const requiredManifestPath = maxBy(
+    manifestFilesForCurrentFolder,
+    (manifestPath) => manifestPath.length
+  );
+
+  if (requiredManifestPath === undefined) {
+    return false;
+  }
+
+  return manifestData[requiredManifestPath].isFlexEnabled;
+}
+
+export async function updateManifestData(
+  manifestUri: string,
+  changeType: FileChangeType
+): Promise<void> {
+  const isFlexEnabled = await readFlexEnabledFlagFromManifestFile(manifestUri);
+  const manifestPath = URI.parse(manifestUri).fsPath;
+  switch (changeType) {
+    case 1: //created
+    case 2: //changed
+      // Parsing of manifest.json failed because the file is invalid
```

Comment:
Change comment to describe the `why`.
We want to keep last successfully read state, because on the manifet.json file may be actively edited.